xrpl/models/base_model.py

```diff
@@ -231,10 +249,21 @@ def from_xrpl(cls: Type[BM], value: Union[str, Dict[str, Any]]) -> BM:
 
         Returns:
             A BaseModel object instantiated from the input.
+
+        Raises:
+            XRPLModelException: If the input dictionary has keys in snake_case format
         """
         if isinstance(value, str):
             value = json.loads(value)
 
+        for k in value:
+            if is_snake_case(k):
+                raise XRPLModelException(
+                    "BaseModel.from_xrpl (and it's overridden) methods accepts JSON "
+                    "keys in the camelCase format only, snake_case inputs are not "
+                    "accepted"
+                )
```

Comment:
This should be done in `_key_to_json` instead, to avoid going through the dictionary twice.